library/test/src/test/java/com/bumptech/glide/load/model/StreamEncoderTest.java

```diff
@@ -47,4 +49,38 @@ public void testWritesDataFromInputStreamToOutputStream() throws IOException {
 
     assertEquals(fakeData, new String(data, "UTF-8"));
   }
+
+  @Test(expected = NullPointerException.class)
+  public void testFileNullException() throws IOException {
+    String fakeData = "SomeRandomFakeData";
+    ByteArrayInputStream is = new ByteArrayInputStream(fakeData.getBytes("UTF-8"));
+    boolean success = false;
+
+    // Write some data to #file making sure file.delete() executes properly
+    encoder.encode(is, file, new Options());
+
+    success = encoder.encode(is, null, new Options());
```

Comment:
If you're just trying to verify that an NPE is thrown when you pass a null File, then I'd use:

```java
assertThrows(NullPointerException.class () -> encoder.encode(is, null, new Options());
```

That way you're verify specific about where the exception is thrown. You'll have to use the actually interface since Glide doesn't support lambdas.

Reply:
Done